projects/storefrontlib/src/cms-components/misc/qualtrics/qualtrics-loader.service.ts

```diff
@@ -1,78 +1,131 @@
+  );
 
-  private setup(): void {
-    const qualtricsScript = this.winRef.document.createElement('script');
-    qualtricsScript.type = 'text/javascript';
-    qualtricsScript.defer = true;
-    qualtricsScript.src = 'assets/qualtricsIntegration.js';
+  constructor(
+    protected winRef: WindowRef,
+    protected rendererFactory: RendererFactory2
+  ) {
+    this.initialize();
+  }
 
-    const idScript = this.winRef.document.createElement('div');
-    idScript.id = this.config.qualtrics.projectId;
+  /**
+   * Starts observing the Qualtrics integration. The integration is based on a
+   * Qualtrics specific event (`qsi_js_loaded`). As soon as this events happens,
+   * we run the API.
+   */
+  protected initialize() {
+    this.qsi$.subscribe(() => this.run());
+  }
 
-    this.winRef.document
-      .getElementsByTagName('head')[0]
-      .appendChild(qualtricsScript);
+  /**
+   * Evaluates the Qualtrics project code for the application.
+   *
+   * In order to reload the evaulation in Qualtrics, the API requires to unload the API before
+   * running it again. We don't do this by default, but offer a flag to conditionally unload the API.
+   */
+  protected run(reload = false): void {
+    if (!this.qsiApi?.API) {
+      if (isDevMode()) {
+        console.log('The QSI api is not available');
+      }
+      return;
+    }
 
-    this.winRef.document.getElementsByTagName('head')[0].appendChild(idScript);
-  }
+    if (reload) {
+      // Removes any currently displaying creatives
+      this.qsiApi.API.unload();
+    }
 
-  private isQualtricsConfigured(): boolean {
-    return (
-      Boolean(this.config.qualtrics) && Boolean(this.config.qualtrics.projectId)
-    );
+    // Starts the intercept code evaluation right after loading the Site Intercept
+    // code for any defined intercepts or creatives
+    this.qsiApi.API.load().done(this.qsiApi.API.run());
   }
 
-  load(): Observable<boolean> {
-    return this.qualtricsLoaded$.pipe(
-      filter((loaded) => loaded),
-      switchMap(() => {
-        const qsi = this.winRef.nativeWindow['QSI'];
-        return this.isDataLoaded().pipe(
-          distinctUntilChanged(),
-          tap((dataLoaded) => {
-            if (dataLoaded) {
-              qsi.API.unload();
-              qsi.API.load().done(qsi.API.run());
-            }
-          })
-        );
-      })
-    );
+  /**
+   * Adds the deployment script to the DOM. If the deployment script is not
+   *
+   * The script will not be added twice if it was loaded before. In that case, we _run_
+   * the Qualtrics API straight away.
+   */
+  addScript(scriptSource: string): void {
+    if (this.hasScript(scriptSource)) {
+      this.run(true);
+    } else {
+      const script: HTMLScriptElement = this.renderer.createElement('script');
+      script.type = 'text/javascript';
+      script.defer = true;
+      script.src = scriptSource;
+      this.renderer.appendChild(this.winRef.document.body, script);
```

Comment:
did the contact get back to your regarding the deployment script needing the `div`

Reply:
nope. Still waiting for that. I'll leave this comment open so we won't forget.

Reply:
We received feedback today from Qualtrics team and we got confirmation that the `DIV` is indeed not needed.